File: LAB-2/BubbleSort.java
import java.util.*;

import java.util.concurrent.TimeUnit;
class BubbleSort
{
    void bubbleSort(int arr[])
    {
        int n = arr.length;
        for (int i = 0; i < n-1; i++)
            for (int j = 0; j < n-i-1; j++)
                if (arr[j] > arr[j+1])
                {
                    
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
    }
 
    
    void printArray(int arr[])
    {
        int n = arr.length;
        for (int i=0; i<n; ++i)
            System.out.print(arr[i] + " ");
        System.out.println();
    }
 
    
    public static void bubblesort()
    {
        Scanner s1 = new Scanner(System.in);
        BubbleSort ob = new BubbleSort();
        int arr[] = {0,0,0,0,0};
        System.out.println("enter array size");
        int n = s1.nextInt();
        System.out.println("enter array elements");
        for (int i=0; i<n; i++)
            arr[i] = s1.nextInt();
        ob.bubbleSort(arr);
        System.out.println("Sorted array");
        ob.printArray(arr);
    }
    
    public static void main(String[] args) 
    {
    	 long startTime = System.nanoTime();
 
   			bubblesort();
 
    	 long endTime = System.nanoTime();
 
    	long durationInNano = (endTime - startTime);  
     
   		 long durationInMillis = TimeUnit.NANOSECONDS.toMillis(durationInNano);
     
   	 	System.out.println(durationInNano);
    	System.out.println(durationInMillis);
    	long timeElapsed = durationInMillis/1000;
    	System.out.println("The execution time in seconds is" +  timeElapsed);
 
	}
}

File: LAB-2/bubblesort.c
#include<stdio.h>
#include<time.h>

void swap(int *x, int *y) 
{ 
    int temp = *x; 
    *x = *y; 
    *y = temp; 
} 

void bubblesort()
{
	int arr[100],i,j,n,temp;
	printf("Enter no. of elements");
	scanf("%d",&n);
	printf("Enter the array elements");
	for(i=0 ; i<n ; i++)
	{
		scanf("%d",&arr[i]);
	}
	for(i=0 ; i < n-1 ; i++)
	{
		for(j=0 ; j < n-i-1 ; j++)
		{
			if (arr[j] > arr[j+1]) 
      		{
        		swap(&arr[j],&arr[j+1]);
        	}
        }
	}
	printf("The sorted array is");
	for(i=0 ; i<n ; i++)
	{
		printf("%d -> ",arr[i]);
	}
}
		
int main()
{
	clock_t start,end; 
    start = clock(); 
    bubblesort(); 
    end = clock(); 
    double time_taken = ((double)(end - start))/CLOCKS_PER_SEC; 
  
    printf("bubblesort() took %f seconds to execute \n", time_taken); 
    return 0; 
}
